custom-menus/non_default_vars_menu.nu

```diff
@@ -0,0 +1,33 @@
+# by default vars_menu displays all the variables, inculding default ones. 
+# this modification is to hide default variables, and leave only those that are set by user.
+
+      {
+        name: vars_menu
+        only_buffer_difference: true
+        marker: "# "
+        type: {
+            layout: list
+            page_size: 10
+        }
+        style: {
+            text: green
+            selected_text: green_reverse
+            description_text: yellow
+        }
+        source: { |buffer, position|
+            $nu.scope.vars
+            | filter {|i| $i.name not-in [
+              "$set_install_state", "$write_default_conf", "$sort_by_date", "$help", "$sort_by_count", 
```

Comment:
I think this is a neat concept. However, I'm not sure what some of these variables are. Are they being set by other scripts you're sourcing?

Some of them I recognize as part of our defaults, others don't seem to be.

Reply:
I don't know the way to get the default ones programmatically, so I just exported all that I had, when I created the new session. I deleted this file for now until I found a way to obtain the default variables automatically.

Reply:
I'm not sure there is a way to get a list of default vars. Really, the only default ones I can think of are `$nu`, `$env`, `$it`, `$in`, and `$nothing`.